ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ValidateMigrationsCommand.java

```diff
@@ -37,12 +55,102 @@
 
   @Override
   protected int command() {
-    throw new UnsupportedOperationException();
+    if (!validateConfigFilePresent()) {
+      return 1;
+    }
+
+    final MigrationConfig config;
+    try {
+      config = MigrationConfig.load(configFile);
+    } catch (KsqlException | MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    return command(
+        config,
+        MigrationsUtil::getKsqlClient,
+        getMigrationsDirFromConfigFile(configFile)
+    );
+  }
+
+  @VisibleForTesting
+  int command(
+      final MigrationConfig config,
+      final Function<MigrationConfig, Client> clientSupplier,
+      final String migrationsDir
+  ) {
+    final Client ksqlClient;
+    try {
+      ksqlClient = clientSupplier.apply(config);
+    } catch (MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    final boolean success;
+    try {
+      success = validate(config, migrationsDir, ksqlClient);
+    } catch (MigrationException e) {
+      LOGGER.error(e.getMessage());
+      return 1;
+    }
+
+    if (success) {
+      LOGGER.info("Successfully validated checksums for migrations that have already been applied");
+      ksqlClient.close();
+    } else {
+      ksqlClient.close();
+      return 1;
+    }
+
+    return 0;
   }
 
   @Override
   protected Logger getLogger() {
     return null;
   }
 
+  /**
+   * @return true if validation passes, else false.
+   */
+  static boolean validate(
```

Comment:
This method is non-private since we'll want to call it from the `apply` command, in order to validate applied migrations prior to submitting any new ones ( cc @jzaralim )